src/utils/useIsomorphicLayoutEffect.ts

```diff
@@ -0,0 +1,3 @@
+import { useEffect, useLayoutEffect } from 'react';
+
+export default typeof document !== 'undefined' ? useLayoutEffect : useEffect;
```

Comment:
what were the issues with using the package directly? the code seeems to be the same? is this to get local npm links working?

Reply:
In our local testing, it was trying to load the CJS build which was causing some issues. We felt it was simplest to just use our own utility here

Reply:
Is there some shared place we can put this? I would rather not copy paste this into multiple repos even though it's a one liner. Is this the only dependency that has this issue? It might be worth spending more time investigating the build infra if this means we can't use third party deps

Reply:
I suspect the issue stems from the fact that the yext sites infra uses node's experimental ESM imports rather than CJS. The useIsomorphicLayoutEffect library defines the ESM with the 'module' property, however that is ignored by node. I expect the issue would be fixed if the library added an 'exports' prop to the package.json file which defines where the ESM module is imported from

Reply:
What is 'exports'? It sounds familiar but I couldn't find it anywhere in the npm docs here https://docs.npmjs.com/cli/v8/configuring-npm/package-json for v6, 7, or 8

Reply:
Ah, I see it here https://nodejs.org/api/packages.html#exports

Reply:
The exports property supports [conditional exports](https://nodejs.org/api/packages.html#conditional-exports) which allows node to use either ESM or CJS. It also allows subpaths to be defined which is what we use in the answers-search-ui package.json file for the i18n builds